### DISCUSSION TITLE
Web Search error

### DISCUSSION THREAD

**Discussion by Ahmed-Mohammed-Eid** [open-webui/open-webui#3811] [Q&A]
I have installed the webui but when I try to use searxng it doesn't work, when I make the url like this 
"http://searxng:8080/search?q=<query>" it gives me this error:
`HTTPConnectionPool(host='searxng', port=8080): Max retries exceeded with url: /search?q=nextjs&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x7fc3eddd5bd0>: Failed to resolve 'searxng' ([Errno -2] Name or service not known)"))
`

and when I change the url to "http://localhost:8080/search?q=<query>"
it gives me another error:
`
Expecting value: line 1 column 1 (char 0)
`

but with the local host it works fine in the browser and the search is working fine and it doesn't work in the webui
![Screenshot 2024-07-12 060632](https://github.com/user-attachments/assets/e58cda76-2c7a-4182-a1c7-2d38f0d6b542)
![Screenshot 2024-07-12 060722](https://github.com/user-attachments/assets/9e18861d-6fc6-4a2c-bf1e-c6352eafe5a4)

**Comment by spencerframe** [comment]
Same exact issue. Please let me know when you find a solution

**Reply by Ahmed-Mohammed-Eid** [reply]
unfortunately I didn't find a solution so I'm using "MSTY" now it has it as a built in feature here's the URL: "https://msty.app/"

**Reply by captainlincom** [reply]
> unfortunately I didn't find a solution so I'm using "MSTY" now it has it as a built in feature here's the URL: "https://msty.app/"

@Ahmed-Mohammed-Eid How do you like MSTY ? Is it easy to use and suits your needs ? Thanks.

**Comment by streetmeat** [comment]
Same issue here. It was previously working as `http://localhost:8080/search?q=<query>` I've made a few updates so I'm not exactly sure when it broke.

**Comment by atasever** [comment]
if you are using open webui within docker change the localhost with host.docker.internal

`http://host.docker.internal:8080/search?q=<query>`

**Reply by xwklwwlf** [reply]
it's work

**Reply by sinister1990** [reply]
I'm getting error: 
403 Client Error: FORBIDDEN for url: http://host.docker.internal:8080/search?q=+weather+in+Zagreb+today&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0

I have set
search:
  safe_search: 0
  autocomplete: ""
  default_lang: ""
  formats:
    - html
    - json
    in settings.yml

**Reply by sinister1990** [reply]
Found the fix.
In searxng container in Files
root: /etc/searxng/settings.yml
Find "formats:"
and add -json and save.
<img width="1268" alt="fix" src="https://github.com/user-attachments/assets/094662b9-3e0d-4144-93b2-67b9810f01d1">

**Reply by CuongNguyen-noob** [reply]
Thank you @atasever so much. I spent weeks to find a solution for this, and here it was, plain and simple.

**Reply by krazor94** [reply]
thank you this worked adding json to the formats and setting the search query to 

`http://searxngIP:8080/search?q=<query>` 

Since i run ollama and openwebUI on a different machine to my searxng which is hosted on my server.

**Comment by nullnuller** [comment]
I followed the [instructions ](https://docs.openwebui.com/tutorial/web_search). It shows that it's searching and shows the URLS, but for some reason recently, it doesn't seem to be sending the web content as context to the LLM. Tried with different LLMs, no luck.

**Reply by mrdevlar** [reply]
Here is the updated link for anyone who is looking:
https://docs.openwebui.com/tutorials/integrations/web_search

**Comment by virtualuk** [comment]
I was having the same problem but then realized that I was running searxng and open-webui in two separate containers (with bridging and port forwarding).  Using 127.0.0.1 (or localhost) wasn't working, I'm presuming because it was trying to connect to the internal localhost of the open-webui docker instance, rather than the host.  As soon as I changed the host to be one of my adapter IP addresses I have web search working with both open-webui and searxng running locally in two separate docker containers.

**Reply by nullnuller** [reply]
In my case, it reports that 4 searches have been made, but for some reason the result is not provided to the LLM as context, so the LLM always complains that there is no web search results. Any reason for this or way to debug it?

**Reply by atasever** [reply]
Sometimes I get this error too. although the search results have the necessary information, when I click on the search result references in open webui, it seems that it cannot access all the information on the referanced page.

**Reply by jockHuxu** [reply]
我也遇到了
现在有解决方案吗

**Comment by alesha-pro** [comment]
Same issue. 
I use this in Kubernetes, and searxng I have as a separate pod in the same namespace. I use a url like `http://servicename.openwebui.svc.cluster.local:port` as for my ollama and pipeline. For ollama and pipeline everything works fine.
I have the same problem - one request at most succeeds, and then Service not known error again.
Can you please tell me what direction to dig in?
I was thinking of opening searxng to public access and providing a link to the public address, but I don't know what to do.

**Comment by iplayfast** [comment]
Searxng is in docker with 8080 as the external port. 

Using http://localhost:8080/search from my browser works.
Using it in WebUI, I get a quick error "expecting value: Line 1 Col1 char(0)

Using http://192.168.1.15:8080/search from my browser also works. 
Using it in WebUI I get
403 Client Error: FORBIDDEN for url: http://192.168.1.15:8080/search?q=2024+US+Presidential+Election+Polls+Latest+Results&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0

I did notice that if I set "bypass ssl for verification" to Yes, save the configuration, go to another tab and then come back it's set to NO.

**Reply by iplayfast** [reply]
I was able to fix it by changing the config file running in the var space. for some reason it ignored my config changes, but worked when I edited the file directly.

**Comment by captainlincom** [comment]
I have disabled the web search , however it still accept my chat prompt with URL. It is weired.
![image](https://github.com/user-attachments/assets/c348d885-955f-4c8a-8b7b-ed039cc2fca7)

![image](https://github.com/user-attachments/assets/0009a27a-e43c-44c4-b0d9-6117a5cf3263)

**Reply by captainlincom** [reply]
I have tried a couple times, it seems that it needs time to let the disable web search to be functioning.
However, when I enable it, it works immediately.

But the LLM output of the webpage summary is completely different from the content of the webpage,
I have tested 5 pages , the only one works well is this page  https://docs.openwebui.com/mission

**Comment by TYTheBeast** [comment]
Running into the same issue with a Searxng instance set up on Hugging Face. What's the fix?

**Comment by GODCREATOR333** [comment]
> if you are using open webui within docker change the localhost with host.docker.internal
> 
> `http://host.docker.internal:8080/search?q=<query>`
Your a life saver. Saved me lot of trouble.
